core/src/cluster_info.rs

```diff
@@ -205,14 +209,9 @@ impl ClusterInfo {
         self.lookup(self.id()).cloned().unwrap()
     }
     fn leader_id(&self) -> Pubkey {
-        let entry = CrdsValueLabel::LeaderId(self.id());
-        self.gossip
-            .crds
-            .lookup(&entry)
-            .and_then(|v| v.leader_id())
-            .map(|x| x.leader_id)
-            .unwrap_or_default()
+        self.gossip_leader_id
     }
+    // Deprecated: don't use leader_data().
     pub fn leader_data(&self) -> Option<&NodeInfo> {
```

Comment:
`leader_data()` is now only used by the banking_stage when [it figures out who to forward transactions to](https://github.com/solana-labs/solana/blob/0be59cad4e4d87329ab08fe64c8f0c83056d11b3/core/src/banking_stage.rs#L100).   Eventually we should be able to refactor things such that the caller of `set_leader()` can just talk directly to the banking_stage and fully remove this from gossip